### PR TITLE
PS2 Ranks and Discord Roles

### PR DESCRIPTION
**Pull request by Normandabald**
Record Discord roles on `discord_user` object.

Also added slash command for manual role syncing (This could be expanded to be a "on Statup" sycing task

image

### CODE REVIEW

File: src/commands/role_added.py [hunk 1]
```diff
@@ -87,5 +134,40 @@ async def on_member_update(self, before: disnake.Member, after: disnake.Member):
...
+        Manually Sync Roles
+        Only available to Guild Administrators
+        """
+        await inter.response.defer(ephemeral=True)
+        await inter.edit_original_message("Syncing roles...")
+        members = inter.guild.members
+        while members:
+            try:
+                member = members.pop()
+                db_record = Database.find_one("members", {'discord_user.id': str(member.id)})
```
Comment (karlis-ozo) [inline src/commands/role_added.py:153]: perhaps run `_add_member_profile` from discord_logger.py here, just in case user joined while bot was restarting.

**Review by karlis-ozo — APPROVED** [review]
it looks like we shoud start to pull some things out of some fies like saving discord member that slipped past us or similar and create something like db_acions or discord_helpers.
I have no clue which way would be best. 
But other than that it looks good

**Comment by Normandabald** [comment]
Latest commit adds the following:
**helpers/sync_commands.py**
Contains a cog with command/sub-commands that sets the framework for future 'sync' jobs.
`/sync discord roles` will do what you'd expect; syncs roles from what is 'real' in Discord to each member in the DB.
Also added support in the `def setup()` in this file to run an initial sync each time the bot is started.

**helpers/discord_db_actions.py**
DiscordDB object class with static methods that `add_role` and `remove_role`.
We can expand on this as we like later on, refactoring other Discord DB jobs into it.

**commands/role_added.py**
A few changes to accomodate the above changes

**src/database_connector.py**
Added extra logging

**loggers/discord_logger.py**
Just general tidying really.
Adjusted some logging, removed unnecessary imports and made some private methods public

**main.py**
Fixed the deprecation warning for `sync_commands_debug`
Added a cog

**Review by karlis-ozo — APPROVED** [review]
was ok, now it's good

**Pull request merged**